Issue 2095: Fixed System Requirements phrasing and added node host info

bfallonf opened the pull request:
@etsauer @timothysc As per #2095 

I'll admit, the part about node hosts is just kinda...blah. I don't think it's saying anything the admin would not expect to do for themselves. Is there anything else that can be added here?

I also changed the wording of the master host section.

Please let me know if this needs adjusting.

etsauer commented:
@bfallonf @timothysc this needs some major updating, as it references the 3.0 docs and (i'm just realizing) has some sections out of order, but this is the guidance the NA Consulting teams give on the subject.

http://playbooks-rhtconsulting.rhcloud.com/playbooks/installation/installation.html#node-sizing

Perhaps we can take some things from here.

timothysc commented on install_config/install/prerequisites.adoc:
```diff
@@ -114,12 +114,20 @@ this during or after installation.
+*Master Hosts*
 
-In general, an {product-title} cluster should have 1 CPU core and 2.5 GB of
+In general, an {product-title} master host should have 1 CPU core and 2.5 GB of
```
In a HA-deployment

bfallonf commented:
@etsauer @timothysc Cool. Added the info I thought was relevant from that 3.0 doc and changed some of the master info to talk about Tim's suggestion.

I'd be hesitant to add more info from that doc. It even talks about how outdated it will be soon.

Thoughts?

etsauer commented on install_config/install/prerequisites.adoc:
```diff
@@ -114,12 +114,27 @@ this during or after installation.
+calculate the expected workload, making sure to leave some room for reserve
+resources, and use this to determine the size of the node host.
+
+In general, add 30% to the maximum expected workload. This will account for 10%
```
The suggestion to add 20% for node failure is for planning capacity for an entire cluster, not for individual nodes. I think the better thing to focus on is that nodes are simply container hosts, and their size should be dictated by the profiles of the workloads expected to run on them. Adding 10% for overhead is still valid though.

bfallonf replied:
@etsauer I read the doc you linked differently. I took the 20% to mean it would be a good size to pick up the slack if other nodes fail. Otherwise, the way you suggest it, what are we adding 20% to? Where does that 20% come from, if not node memory or whatever?

etsauer replied:
@bfallonf yes that's true, the 20% is meant for picking up slack in case a node fails. It just sounds a bit strange when we're talking about individual nodes, seemingly suggesting that 20% of a node could fail somehow. That figure also makes the assumption that there are at least 5 schedulable nodes in the cluster. I think in this context, when we're trying to be brief it makes more sense to just say calculate expected maximum workload per node, add 10% for overhead, and for production deployments, make sure to allocate enough resources such that a host failure doesn't restrict your maximum capacity.

bfallonf commented:
@etsauer Thanks for the additional help. Changed to:

"The size of a node host depends on the expected size of its workload. As an {product-title} cluster administrator, you will need to calculate the expected workload, then add about 10% for overhead. For production environments, allocate enough resources so that node host failure does not affect your maximum capacity."

WDYT?

etsauer commented:
lgtm

bfallonf commented:
@etsauer Cool. Thanks. I'll move forward with this.

@tpoitras @tnguyen-rh @ahardin-rh @adellape Peer review, please?

bfallonf commented:
[rev_history]
|link:../install_config/install/prerequisites.html[Installing -> Prerequisites]
|Added prerequisite information to node host section of link:../install_config/install/prerequisites.html#system-requirements[System Requirements].
%

tnguyen-rh commented:
@bfallonf LGTM.

bfallonf commented:
@tnguyen-rh Great. Thanks. I'll merge.
